### PR TITLE
Recreate missing java.io.tmpdir when creating LDD download temp files

### PR DESCRIPTION
**Pull request by jordanpadams**
## 🗒️ Summary
`SchemaUpdater.createLddTempFile()` downloads an LDD to a temp file, trying `Files.createTempFile()` with POSIX permissions first and falling back to plain `File.createTempFile()` on failure. Both calls use the JVM's default `java.io.tmpdir` with no directory override, so on a host where that directory doesn't exist, both attempts fail identically with `IOException: The system cannot find the path specified` — and every single product that needs an LDD update fails.

This was reported against Harvest 5.1.3 on a Windows host: a bundle with 7,834 products failed 100% (0 loaded) because of this one systemic temp-directory issue, even though the referenced LDD schema itself was valid and reachable.

Fix: before attempting temp file creation, ensure the resolved `java.io.tmpdir` directory exists (`mkdirs()`), and include the resolved path in the `LddException` message if creation still fails, so the failure is actionable.

### 🤖 AI Assistance Disclosure
- [ ] No AI assistance used
- [ ] AI used for light assistance (e.g., suggestions, refactoring, documentation help, minor edits)
- [ ] AI used for moderate content generation (AI generated some code or logic, but the developer authored or heavily revised the majority)
- [x] AI generated substantial portions of this code

Estimated % of code influenced by AI: 90 %

## ⚙️ Test Data and/or Report
Added `createLddTempFile_recreatesMissingTmpDir` to `TestSchemaUpdaterTempFile`, which points `java.io.tmpdir` at a directory that does not yet exist and asserts `createLddTempFile()` creates it and succeeds. Ran locally:

```
mvn test -Dtest=TestSchemaUpdaterTempFile
...
Tests run: 2, Failures: 0, Errors: 0, Skipped: 0
BUILD SUCCESS
```

## ♻️ Related Issues
Refs NASA-PDS/registry-loader#142

## 🤓 Reviewer Checklist

*Reviewers: Please verify the following before approving this pull request.*

### Documentation and PR Content
- [ ] **Documentation:** README, Wiki, or inline documentation (Sphinx, Javadoc, Docstrings) have been updated to reflect these changes.
- [ ] **Issue Traceability:** The PR is linked to a valid GitHub Issue
- [ ] **PR Title:** The PR title is "user-friendly" clearly identifying what is being fixed or the new feature being added, that if you saw it in the Release Notes for a tool, you would be able to get the gist of what was done.

### Security & Quality
- [ ] **SonarCloud:** Confirmed no new High or Critical security findings.
- [ ] **Secrets Detection:** Verified that the Secrets Detection scan passed and no sensitive information (keys, tokens, PII) is exposed.
- [ ] **Code Quality:** Code follows organization style guidelines and best practices for the specific language (e.g., PEP 8, Google Java Style).

### Testing & Validation
- [ ] **Test Accuracy:** Verified that test data is accurate, representative of real-world PDS4 scenarios, and sufficient for the logic being tested.
- [ ] **Coverage:** Automated tests cover new logic and edge cases.
- [ ] **Local Verification:** (If applicable) Successfully built and ran the changes in a local or staging environment.

### Maintenance
- [ ] **Backward Compatibility:** Confirmed that these changes do not break existing downstream dependencies or API contracts (or that breaking changes are clearly documented).

### CODE REVIEW

File: src/main/java/gov/nasa/pds/registry/common/es/service/SchemaUpdater.java [hunk 1]
```diff
@@ -273,6 +273,12 @@ private String getJsonUrl(String uri) {
    * UnsupportedOperationException or IOException when POSIX attributes are applied.
    */
   public static File createLddTempFile(String prefix) throws LddException {
+    String tmpDirPath = System.getProperty("java.io.tmpdir");
+    File tmpDir = new File(tmpDirPath);
+    if (!tmpDir.exists()) {
+      tmpDir.mkdirs();
```
Comment (nutjob4life) [inline src/main/java/gov/nasa/pds/registry/common/es/service/SchemaUpdater.java:279]: This returns a boolean and `false` can indicate failure, but this is _so highly unlikely_ given that `java.io.tmpir` is almost always a writeable location. I wouldn't bother, but I am a pedant as you know 🤭

**Review by nutjob4life — APPROVED** [review]
Pedant comment: added
Code delta: confirmed
Test case: actually quite good
Maven tests: passing
Overall approval: ✅
Maven details:
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running dao.TestDataLoaderIgnoreConflicts
2026-08-10T13:10:29.637905Z main ERROR LIDVID = urn:test::1.0, Message = conflict
2026-08-10T13:10:29.651540Z main ERROR LIDVID = urn:test::1.0, Message = conflict
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.545 s -- in dao.TestDataLoaderIgnoreConflicts
[INFO] Running meta.TestAutogenExtractor
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.187 s -- in meta.TestAutogenExtractor
[INFO] Running parser.TestClassAttrAssociationParser
[INFO] Tests run: 6, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.284 s -- in parser.TestClassAttrAssociationParser
[INFO] Running service.TestJsonLddLoaderOverwrite
[INFO] Tests run: 7, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.309 s -- in service.TestJsonLddLoaderOverwrite
[INFO] Running service.TestSchemaUpdaterTempFile
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.039 s -- in service.TestSchemaUpdaterTempFile
[INFO] Running service.TestLddUrlUtils
[INFO] Tests run: 4, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.050 s -- in service.TestLddUrlUtils
[INFO] Running tt.TestLddFieldResolution
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 1.720 s -- in tt.TestLddFieldResolution
[INFO] Running tt.TestDateFormats
                          null  -->  null
                                -->  null
          2013-10-24T00:00:00Z  -->  2013-10-24T00:00:00Z
      2013-10-24T00:49:37.457Z  -->  2013-10-24T00:49:37.457Z
         2013-302T01:02:03.123  -->  2013-10-29T01:02:03.123Z
        2013-302T01:02:03.123Z  -->  2013-10-29T01:02:03.123Z
                   2016-09-08Z  -->  2016-09-08T00:00:00Z
                    2013-03-02  -->  2013-03-02T00:00:00Z
                       2013-12  -->  2013-12-01T00:00:00Z
                          2013  -->  2013-01-01T00:00:00Z
                         2015Z  -->  2015-01-01T00:00:00Z
                      2013-001  -->  2013-01-01T00:00:00Z
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.028 s -- in tt.TestDateFormats
[INFO] Running gov.nasa.pds.registry.common.es.service.TestSearchIndexWait
[INFO] Tests run: 10, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 2.070 s -- in gov.nasa.pds.registry.common.es.service.TestSearchIndexWait
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 39, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
```
